plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/wizard/IKeyValueWizardModel.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors: Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.common.ui.wizard;
+
+import org.eclipse.core.databinding.validation.IValidator;
+
+public interface IKeyValueWizardModel<T extends IKeyValueItem> {
+	String PROPERTY_KEY = "key";
+	String PROPERTY_VALUE = "value";
+	
+	String getKey();
+	void setKey(String key);
+	
+	String getValue();
+	void setValue(String value);
+	
+	String getTitle();
+	String getDescription();
+	
+	/**
+	 * Retrieve the validator to be applied after the key is converted
+	 * @return
+	 */
+	IValidator getKeyAfterConvertValidator();
```

Comment:
@adietish The only purpose is so the view can retrieve the validator thus making the view and model generic enough to handle labels or environment variables